### PR TITLE
#42 move bucketfs location

### PR DESCRIPTION
**Pull request by MarleneKress79789**
First part of https://github.com/exasol/data-science-utils-python/issues/46
Added bucketfs location files.

### CODE REVIEW

File: doc/changes/changes_0.1.0.md [hunk 1]
```diff
@@ -13,3 +13,4 @@ Code name: Initial implementation
 ### Refactorings
 
   - #15: Remove DepHell dependency, because it is not maintained anymore
+  - #46: Moved BucketFS Location over from exasol_data_science_utils_python
```
Comment (tomuben) [inline doc/changes/changes_0.1.0.md:16]: I created a new ticket in this repo: #42. we should use that one for the history.

File: exasol_bucketfs_utils_python/abstract_bucketfs_location.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import typing
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class AbstractBucketFSLocation(ABC):
+
```
Comment (tomuben) [inline exasol_bucketfs_utils_python/abstract_bucketfs_location.py:7]: Add a short comment what this class does.

File: exasol_bucketfs_utils_python/bucketfs_factory.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+from exasol_bucketfs_utils_python.bucket_config import BucketConfig
+from exasol_bucketfs_utils_python.bucketfs_config import BucketFSConfig
+from exasol_bucketfs_utils_python.bucketfs_connection_config import BucketFSConnectionConfig
+
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+from exasol_bucketfs_utils_python.localfs_mock_bucketfs_location import LocalFSMockBucketFSLocation
+
+
+class BucketFSFactory:
+
```
Comment (tomuben) [inline exasol_bucketfs_utils_python/bucketfs_factory.py:14]: Add a short comment what this class does.

File: exasol_bucketfs_utils_python/bucketfs_location.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+from typing import Any
+
+from exasol_bucketfs_utils_python import download, upload
+from exasol_bucketfs_utils_python.bucket_config import BucketConfig
+
+from exasol_bucketfs_utils_python.abstract_bucketfs_location import AbstractBucketFSLocation
+
+
+class BucketFSLocation(AbstractBucketFSLocation):
+
```
Comment (tomuben) [inline exasol_bucketfs_utils_python/bucketfs_location.py:12]: Add a short comment what this class does.

File: exasol_bucketfs_utils_python/bucketfs_location.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    def __init__(self, bucket_config: BucketConfig, base_path: PurePosixPath):
+        self.base_path = base_path
+        self.bucket_config = bucket_config
+
+    def get_complete_file_path_in_bucket(self, bucket_file_path) -> str:
```
Comment (tomuben) [inline exasol_bucketfs_utils_python/bucketfs_location.py:17]:
```suggestion
    def get_complete_file_path_in_bucket(self, bucket_file_path: str) -> str:
```

File: exasol_bucketfs_utils_python/bucketfs_factory.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    def create_bucketfs_location(self, url: str, user: str, pwd: str, base_path: Union[PurePosixPath, None] = None):
```
Comment (tomuben) [inline exasol_bucketfs_utils_python/bucketfs_factory.py:15]:
```suggestion
    def create_bucketfs_location(self, url: str, user: str, pwd: str, base_path: Optional[PurePosixPath] = None):
```

File: exasol_bucketfs_utils_python/localfs_mock_bucketfs_location.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+from pathlib import PurePosixPath, Path
+from typing import Any
+
+import joblib
+
+from exasol_bucketfs_utils_python.abstract_bucketfs_location import AbstractBucketFSLocation
+
+
+class LocalFSMockBucketFSLocation(AbstractBucketFSLocation):
+
```
Comment (tomuben) [inline exasol_bucketfs_utils_python/localfs_mock_bucketfs_location.py:11]: Add a short comment what this class does.

File: doc/changes/changes_0.1.0.md (hunk 1 above)
Comment (MarleneKress79789) [reply doc/changes/changes_0.1.0.md:16]: Soll ich den Branch dem entsprechend auch umbenennen?

**Review by tomuben — APPROVED** [review]

File: doc/changes/changes_0.1.0.md (hunk 1 above)
Comment (tomuben) [reply doc/changes/changes_0.1.0.md]: Not necessary, we usually delete the branch after integration into main.

**Pull request merged**